WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java

```diff
@@ -0,0 +1,753 @@
+            public void onClick(DialogInterface dialog, int which) {
+                switch (which) {
+                    case DialogInterface.BUTTON_NEGATIVE:
+                        undoChanges();
+                        break;
+                    case DialogInterface.BUTTON_POSITIVE:
+                        updateAccountOrContinue();
+                        break;
+                    // DialogInterface.BUTTON_NEUTRAL is intentionally ignored.  Just dismiss dialog.
+                }
+            }
+        };
+
+        AlertDialog dialog = new AlertDialog.Builder(new ContextThemeWrapper(getActivity(), R.style.LoginTheme))
+                .setMessage(message)
+                .setNeutralButton(R.string.login_error_button, dialogListener)
+                .setNegativeButton(R.string.signup_epilogue_error_button_negative, dialogListener)
+                .setPositiveButton(R.string.signup_epilogue_error_button_positive, dialogListener)
+                .create();
+        dialog.show();
+    }
+
+    protected void showErrorDialogAvatar() {
+        AlertDialog dialog = new AlertDialog.Builder(new ContextThemeWrapper(getActivity(), R.style.LoginTheme))
+                .setMessage(R.string.signup_epilogue_error_avatar)
+                .setPositiveButton(R.string.login_error_button, null)
+                .create();
+        dialog.show();
+    }
+
+    protected void startCropActivity(Uri uri) {
+        final Context context = getActivity();
+
+        if (context != null) {
+            UCrop.Options options = new UCrop.Options();
+            options.setShowCropGrid(false);
+            options.setStatusBarColor(ContextCompat.getColor(context, R.color.status_bar_tint));
+            options.setToolbarColor(ContextCompat.getColor(context, R.color.color_primary));
+            options.setAllowedGestures(UCropActivity.SCALE, UCropActivity.NONE, UCropActivity.NONE);
+            options.setHideBottomControls(true);
+
+            UCrop.of(uri, Uri.fromFile(new File(context.getCacheDir(), "cropped.jpg")))
+                    .withAspectRatio(1, 1)
+                    .withOptions(options)
+                    .start(getActivity(), this);
+        }
+    }
+
+    protected void startGravatarUpload(final String filePath) {
+        if (!TextUtils.isEmpty(filePath)) {
+            final File file = new File(filePath);
+
+            if (file.exists()) {
+                startProgress();
+
+                GravatarApi.uploadGravatar(file, mAccountStore.getAccount().getEmail(), mAccountStore.getAccessToken(),
+                        new GravatarApi.GravatarUploadListener() {
+                            @Override
+                            public void onSuccess() {
+                                endProgress();
+                                mPhotoUrl = GravatarUtils.fixGravatarUrl(mAccount.getAccount().getAvatarUrl(),
+                                        getResources().getDimensionPixelSize(R.dimen.avatar_sz_login_epilogue));
```

Comment:
Good call incorporating the `injectCache()` track @theck13 , I can see the new image be used immediately on the epilogue after setting it.

Here's a thingie: turns out that the injected image response is for a URL that is slightly different to the URL loaded in the MeFragment after the epilogue is dimissed, which causes the annoyance where the Me screen doesn't load the injected avatar.

Problem seems to be the size parameter which is different between these two screens. What do you think about handling this issue in this PR?

Reply:
Nice catch!  I updated the size in `SignupEpilogueFragment.java` to match `MeFragment.java` in c75f5a7a8a6de7a5c16abcf730bda63d2090aee9.

Reply:
Works like a charm now, thanks! 👍